### PR TITLE
Adjust helm chart to be used with the config-controller.

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/16589

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/cert-operator/templates/configmap.yaml
Comment (stone-z) [inline helm/cert-operator/templates/configmap.yaml:16]: I've never interacted with this value before -- what does it do? Or at least, could the change affect which CRs this version of `cert-operator` will reconcile?

File: helm/cert-operator/Chart.yaml
Comment (kubasobon) [reply helm/cert-operator/Chart.yaml:11]: https://intranet.giantswarm.io/docs/dev-and-releng/configuration-management/

Comment (kubasobon) [reply helm/cert-operator/Chart.yaml:11]: TL;DR: Set version annotation to a branch or major version (e.g. `1.x.x`) here. `config-controller` will use that information when generating ConfigMap and Secret by pulling matching `giantswarm/config` version (branch/tag).

Comment (stone-z) [reply helm/cert-operator/Chart.yaml:11]: I see, and here `cert-operator` is actually the name of the branch of the `config` repo -- thanks!

Comment (corest) [reply helm/cert-operator/Chart.yaml:11]: https://intranet.giantswarm.io/docs/dev-and-releng/configuration-management/

File: helm/cert-operator/templates/configmap.yaml
Comment (corest) [reply helm/cert-operator/templates/configmap.yaml:16]: It is empty everywhere. AFAIK that's some legacy unused stuff. But I'll make it configurable in case we need to change it

File: helm/cert-operator/ci/default-values.yaml
Comment (kopiczko) [inline helm/cert-operator/ci/default-values.yaml:63]: So almost all of that wasn't used?

**Review by kopiczko — APPROVED** [review]
What is with this title?

File: helm/cert-operator/Chart.yaml
Comment (corest) [reply helm/cert-operator/Chart.yaml]: `cert-operator` used here is just for testing and will be updated before merging

File: helm/cert-operator/ci/default-values.yaml
Comment (corest) [reply helm/cert-operator/ci/default-values.yaml:63]: yes

**Review by MarcelMue — APPROVED** [review]
LGTM

**Review by stone-z — APPROVED** [review]
LGTM once tested and test value is reverted

File: helm/cert-operator/ci/default-values.yaml
Comment (kopiczko) [reply helm/cert-operator/ci/default-values.yaml:63]: ...

**Pull request merged**
